ft_strncmp.c

```diff
@@ -0,0 +1,36 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_strncmp.c                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: hbalan <hbalan@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/01/13 14:49:14 by hbalan            #+#    #+#             */
+/*   Updated: 2023/01/13 14:49:16 by hbalan           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+int	ft_strncmp(const char *s1, const char *s2, size_t n)
+{
+	size_t	i;
+
+	i = 0;
+	while ((s1[i] || s2[i]) && (i < n))
+	{
+		return (((unsigned char)s1[i] - (unsigned char)s2[i]));
+		i++;
+	}
+	return (0);
+}
+
```

Comment:
While testing this case:
```
char str[100] = "axarningisfun";
char str2[100] = "a";
printf("%d\n", ft_strncmp(str, str2, 2));
printf("%d", strncmp(str, str2, 2));
```

the result is:
```
0
120
```

In the following case:
```
char str[100] = "axarningisfun";
char str2[100] = "auioyt";
printf("%d\n", ft_strncmp(str, str2, 2));
printf("%d", strncmp(str, str2, 2));
```
	
the result is:
```
0
3
```

Looks like if the first character matches function doesn't not go any further, even if "n" allows to do so.
The correct behaviour should be: the **first character that does not match has a lower value in str1 than in str2** returns < 0 and **the first character that does not match has a greater value in str1 than in str2**. returns > 0. So you should loop, until you find a non matching symbol, if I got it right.

This could be an answer. Please re-check me
```
int	ft_strncmp(const char *s1, const char *s2, size_t n)
{
	size_t	i;
	size_t	buff;

	i = 0;
	while ((s1[i] || s2[i]) && (i < n))
	{	
		buff = (unsigned char)s1[i] - (unsigned char)s2[i];
		if(buff != 0)
		{
			return (buff);
		}
		i++;
		
	}
	return (0);
}

```

Reply:
Thank you for notice! You're right, I've missed this condition (